apps/root/src/common/components/transaction-steps/index.tsx

```diff
@@ -333,7 +407,7 @@ const buildApproveTokenItem = ({
               >
                 {hasPendingApproval ? waitingForAppvText : isPermit2Enabled ? infiniteBtnText : specificBtnText}
               </Button>
-              {!isPermit2Enabled && (
+              {isPermit2Enabled && (
```

Comment:
was this a bug that we had?

Reply:
Indeed, after testing this once again, I found out that this button wasn't rendering in the right conditions.
Now, this secondary button (exact amount approval) is only rendered when permit2 is enabled

Reply:
Exact approval should actually appear when permit2 is disabled, now that I'm thinking about it we should always show exact approvals

Reply:
Agree, maybe I didn't explain myself clearly, I believe this is implemented as expected:

### Permit2 Enabled
![Screenshot 2024-02-08 at 16 52 10](https://github.com/Mean-Finance/dca-fe/assets/144699868/02e77910-c849-48a3-b65e-0db7d00d4371)

### Permit2 Disabled
![Screenshot 2024-02-08 at 16 55 18](https://github.com/Mean-Finance/dca-fe/assets/144699868/7c575a2d-4d69-4c88-b114-69cd83a3ff78)

Reply:
Gotcha